packages/blockchain-wallet-v4-frontend/src/data/modules/coinify/sagas.js

```diff
@@ -82,38 +80,45 @@ export default ({ coreSagas }) => {
       yield put(A.coinifyLoading())
       const trade = yield call(coreSagas.data.coinify.sell)
 
+      const state = yield select()
+
       if (!trade) {
         const trade = yield select(selectors.core.data.coinify.getTrade)
         const parsed = JSON.parse(trade.error)
 
         yield put(A.coinifyFailure(parsed))
         return
       }
-      const p = yield select(sendBtcSelectors.getPayment)
+      const p = path(['coinify', 'payment'], state)
       let payment = yield coreSagas.payment.btc.create(
         { payment: p.getOrElse({}), network: settings.NETWORK })
       payment = yield payment.amount(parseInt(trade.sendAmount))
 
       // QA Tool: manually set a "to" address on the payment object for testing sell
-      const qaState = yield select()
-      const qaAddress = path(['qa', 'qaSellAddress'], qaState)
-
+      const qaAddress = path(['qa', 'qaSellAddress'], state)
       if (qaAddress) {
         payment = yield payment.to(qaAddress)
       } else {
         payment = yield payment.to(path(['transferIn', 'details', 'account'], trade))
       }
 
-      payment = yield payment.description(`Exchange Trade COINIFY=${trade.id}`)
-      payment = yield payment.build()
+      payment = yield payment.description(`Exchange Trade CNY-${trade.id}`)
+      try {
+        payment = yield payment.build()
```

Comment:
payment.build() can fail due to missing 'to', wrong fee and similar field values, should we really continue sending in such cases?

Reply:
The only way 'to' will be missing is if the trade fails, in which case there would be an error and we shouldn't make it this far. The fee will always be the priority fee.